libbeat/cmd/keystore.go

```diff
@@ -0,0 +1,230 @@
+package cmd
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	tml "golang.org/x/crypto/ssh/terminal"
+
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/cmd/instance"
+	"github.com/elastic/beats/libbeat/common/terminal"
+	"github.com/elastic/beats/libbeat/keystore"
+)
+
+func getKeystore(name, version string) keystore.Keystore {
+	b, err := instance.NewBeat(name, "", version)
+
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error initializing beat: %s\n", err)
+		os.Exit(1)
+	}
+
+	if err = b.Init(); err != nil {
+		fmt.Fprintf(os.Stderr, "Error initializing beat: %s\n", err)
+		os.Exit(1)
+	}
+
+	return b.Keystore()
+}
+
+// genKeystoreCmd initialize the Keystore command to manage the Keystore
+// with the following subcommands:
+//  - create
+//  - add
+//  - remove
+//  - list
+func genKeystoreCmd(name, idxPrefix, version string, beatCreator beat.Creator, runFlags *pflag.FlagSet) *cobra.Command {
+	keystoreCmd := cobra.Command{
+		Use:   "keystore",
+		Short: "Manage secrets keystore",
+	}
+
+	keystoreCmd.AddCommand(genCreateKeystoreCmd(name, version))
+	keystoreCmd.AddCommand(genAddKeystoreCmd(name, version))
+	keystoreCmd.AddCommand(genRemoveKeystoreCmd(name, version))
+	keystoreCmd.AddCommand(genListKeystoreCmd(name, version))
+
+	return &keystoreCmd
+}
+
+func genCreateKeystoreCmd(name, version string) *cobra.Command {
+	command := &cobra.Command{
+		Use:   "create",
+		Short: "Create keystore",
+		Run: func(cmd *cobra.Command, args []string) {
```

Comment:
another option to decouple the `cobra.Command` from the function:

```
func genCreateKeystoreCmd(name, version string) *cobra.Command {
  var force bool
  cmd := &cobra.Command{
    Use: ...
    Short: ...
    Run: func(cmd *cobra.Command, args []string) {
      if err := runCreateKeystore(name, version, force); err != nil {
        fmt.Error(err)
      }
    }
  }
  cmd.Flags().BoolVar(&force, "force", false, "...")

  return cmd
}
```

Common options can be added to the parent-command by using PersistentFlags().